src/edge_detection.cpp

```diff
@@ -26,28 +26,28 @@ namespace
     template<typename _Type>
     void createPositiveXEdge( const std::vector< _Type > & data, std::vector < PointBase2D<_Type> > & point, _Type x, _Type y )
     {
-        for ( std::vector < _Type >::const_iterator dataX = data.cbegin(); dataX != data.cend(); ++dataX )
+        for ( typename std::vector < _Type >::const_iterator dataX = data.cbegin(); dataX != data.cend(); ++dataX )
             point.push_back( PointBase2D<_Type>( (*dataX) + x, y ) );
     }
 
     template<typename _Type>
     void createNegativeXEdge( const std::vector< _Type > & data, std::vector < PointBase2D<_Type> > & point, _Type x, _Type y )
     {
-        for ( std::vector <_Type>::const_iterator dataX = data.cbegin(); dataX != data.cend(); ++dataX )
+        for ( typename std::vector <_Type>::const_iterator dataX = data.cbegin(); dataX != data.cend(); ++dataX )

```

Comment:
Just out of curiosity, for self-education, why do we need `typename` keyword here?

Reply:
`std::vector<>` is a template class and then we pass a template type to it so some compilers cannot understand it. Visual Studio compiler understands such while **g++** doesn't.

Reply:
I definitely learned something new today. Thank you very much!